### ISSUE TITLE
llvm --without-rtti fails to build

### ISSUE THREAD

**Issue by apjanke** [Homebrew/homebrew-core#4400]
`brew install llvm --without-rtti` errors out at the configuration stage.

```
$ brew install -s llvm --without-rtti
==> Using the sandbox
...
CMake Error at cmake/modules/HandleLLVMOptions.cmake:627 (message):
  Exception handling requires RTTI.  You must set LLVM_ENABLE_RTTI to ON
Call Stack (most recent call first):
  CMakeLists.txt:415 (include)
```

https://gist.github.com/64db6e1cf71d0ecc09438f8b9809bc90

Looks like this is because we're enabling exception handling unconditionally, but it requires RTTI. So exception handling should probably be turned off as part of the `--without-rtti` option.

**Comment by apjanke** [comment]
Looks like this was introduced by #3028, which enabled exception handling along with the 3.8.1 upgrade.

**Comment by apjanke** [comment]
Fixed by #4422.
